encoder.lua

```diff
@@ -210,15 +246,72 @@ local create_audio = function(start_timestamp, end_timestamp, filename, padding)
     end
 end
 
+-- Calls the proper function depending on whether or not the snapshot should be animated
+local create_video_media = function(start_timestamp, end_timestamp, timestamp, filename)
+    if not h.is_empty(self.config.image_field) then
+        local source_path = mp.get_property("path")
+        local output_path = utils.join_path(self.platform.tmp_dir(), filename)
+
+        local on_finish = function()
+            self.store_fn(filename, output_path)
+            os.remove(output_path)
+        end
+
+        if self.video.animation_enabled then 
+            create_animated_snapshot(start_timestamp, end_timestamp, source_path, output_path, on_finish)
+        else 
+            create_snapshot(timestamp, source_path, output_path, on_finish)
+        end
+    else
+        print("Video media will not be created.")
+    end
+end
+
+-- Generate a filename for the video, taking care of its extension and whether it's an animation or a snapshot
+local make_video_filename = function(start_time, end_time, timestamp)
+    if self.video.animation_enabled then
+        return filename_factory.make_animated_snapshot_filename(start_time, end_time, self.video.extension)
+    else
+        return filename_factory.make_snapshot_filename(timestamp, self.video.extension)
+    end
+end
+
+-- Generates a filename for the audio
+local make_audio_filename = function(start_time, end_time) 
+    return filename_factory.make_audio_filename(start_time, end_time, self.audio.extension) 
+end
+
+-- Toggles on and off animated snapshot generation at runtime. It is called whenever ctrl+g is pressed
+local toggle_animation = function()
+    self.video.animation_enabled = not self.video.animation_enabled
+    self.video.extension = self.video.animation_enabled and self.config.animated_snapshot_extension or self.config.snapshot_extension
+    mp.osd_message("Animation " .. (self.video.animation_enabled and "enabled" or "disabled"))
+end
+
+-- Sets the module to its preconfigured status
 local init = function(config, store_fn, platform)
     self.config = config
     self.store_fn = store_fn
     self.platform = platform
     self.encoder = config.use_ffmpeg and ffmpeg or mpv
+
+    self.video.animation_enabled = config.animated_snapshot_enabled
+    self.video.extension = config.animated_snapshot_enabled and config.animated_snapshot_extension or self.config.snapshot_extension
+
+    self.audio.extension = self.config.audio_extension
```

Comment:
`self.audio.extension` and `self.video.extension` aren't really needed because you're passing their own extensions to each function respectfully.

Reply:
`self.video.extension` is really just equivalent to writing `self.video.animation_enabled and self.config.animated_snapshot_extension or self.config.snapshot_extension`.
The point is it gets updated at runtime whenever animation is toggled on and off, so the conditional expression can be avoided whenever the extension is used in any function (by just writing self.video.extension).
Currently, though the extension is used only in one function, which is the make_video_filename function, so I don't mind replacing self.video.animation_enabled with the conditional expression.

The other one, `self.audio.extension` was added only for the purpose of simmetry, so I'd agree that it's currently useless.

As for `self.video.animation_enabled`, I followed the same pattern used for the clipboard autocopy, namely it gets assigned the config value in the init function and then gets toggled on and off at runtime. I would keep it as it is honestly.